### PR TITLE
Remove SSL support from .NET, use non-root user

### PR DESCRIPTION
**Pull request by bwateratmsft**
This PR removes SSL support for ASP.NET apps. It also switches to a non-root user in the Dockerfile, if all the ports are >1024 (same behavior as Python).

### CODE REVIEW

File: resources/templates/netCore/Dockerfile.template [hunk 1]
```diff
@@ -4,6 +4,18 @@ WORKDIR /app
...
 {{/each}}
 
+{{#if (eq platform '.NET: ASP.NET Core')}}
+{{#if ports}}
+ENV ASPNETCORE_URLS=http://+:{{ ports.[0] }}
+{{/if}}
+{{/if}}
+
+{{#unless (isRootPort ports)}}
+# Switching to a non-root user, please refer to https://aka.ms/vscode-docker-python-user-rights
```
Comment (bwateratmsft) [inline resources/templates/netCore/Dockerfile.template:14]: TODO: It says "Python" (so do the docs)

File: src/tasks/netcore/NetCoreTaskHelper.ts [hunk 2]
```diff
@@ -275,43 +218,10 @@ export class NetCoreTaskHelper implements TaskHelper {
...
             addVolumeWithoutConflicts(volumes, appVolume);
             addVolumeWithoutConflicts(volumes, srcVolume);
             addVolumeWithoutConflicts(volumes, debuggerVolume);
             addVolumeWithoutConflicts(volumes, nugetVolume);
-            if (await fse.pathExists(nugetFallbackVolume.localPath)) {
-                addVolumeWithoutConflicts(volumes, nugetFallbackVolume);
-            }
-        }
-
-        if (userSecrets || ssl) {
```
Comment (bwateratmsft) [inline src/tasks/netcore/NetCoreTaskHelper.ts:294]: Do we want to stop mapping the user secrets folder? (Users can manually map it of course)

Comment (karolz-ms) [reply src/tasks/netcore/NetCoreTaskHelper.ts:294]: If we stop mapping, will that break users that scaffolded the app with SSL (using previous version of the extension)?

**Review by karolz-ms — APPROVED** [review]

**Comment by karolz-ms** [comment]
Soo glad to see this finally happening! ☺️

**Comment by bwateratmsft** [comment]
> Soo glad to see this finally happening! ☺️

Same. I agree with @BigMorty though that we owe a meeting with the ASP.NET folks to see what they think, before we merge this.

File: src/tasks/netcore/NetCoreTaskHelper.ts (hunk 2 above)
Comment (bwateratmsft) [reply src/tasks/netcore/NetCoreTaskHelper.ts:294]: 🤦‍♂️ Yeah, I should probably try that out.

**Comment by bwateratmsft** [comment]
Closing this since we're changing directions here.

**Pull request closed without merging**